rllib/evaluation/collectors/agent_collector.py

```diff
@@ -191,7 +195,19 @@ def add_action_reward_next_obs(self, input_values: Dict[str, TensorType]) -> Non
 
         # Default to next timestep if not provided in input values
         if SampleBatch.T not in input_values:
+            if log_once("no_timesteps_provided_in_add_action_reward_next_obs"):
+                logger.warning(
+                    "Agent collector is not receiving timesteps."
+                    "Timesteps are built internally."
+                )
             values[SampleBatch.T] = self.buffers[SampleBatch.T][0][-1] + 1
+        if SampleBatch.REWARDS not in input_values:
```

Comment:
During inference, it's common that we simply don't have rewards at hand. So set them to zero by default. @kouroshHakha

Reply:
(This should be handled by the used_for_compute_actions flag. But our current way of doing things is that SampleBatch.PREV_REWARDS is used_for_compute_actions in our default view requirements)